src/Packagist/WebBundle/Controller/PackageController.php

```diff
@@ -506,7 +508,7 @@ public function viewPackageVersionAction(Request $req, $versionId)
 
         /** @var VersionRepository $repo  */
         $repo = $this->getDoctrine()->getRepository('PackagistWebBundle:Version');
-        if (!$this->isGranted('ROLE_MAINTAINER', $repo->find($versionId))) {
+        if (!$this->isGranted('ROLE_MAINTAINER', $repo->find($versionId) && !$this->allowReadOnlyAccess())) {
```

Comment:
if (!$this->isGranted('ROLE_MAINTAINER', $repo->find($versionId)) && !$this->allowReadOnlyAccess()) {